docs/rules/no-namespace.md

```diff
@@ -0,0 +1,95 @@
+# Disallows the use of `internal modules` and `namespaces`.
```

Comment:
Could we please call them "custom TypeScript modules" just to be 100% clear in differentiating them from ES2015 modules. "Internal" feels too vague here.